hoomd/hpmc/pytest/test_shape.py

```diff
@@ -0,0 +1,282 @@
+import hoomd
+import hoomd.hpmc
+from hoomd.hpmc import _hpmc
+import numpy as np
+import pytest
+from hoomd.hpmc.pytest.conftest import *
+
+
+def check_dict(shape_dict, args):
+    """
+    check_dict: Function to check that two dictionaries are equivalent
+
+    Arguments: shape_dict and args - dictionaries to test
+
+    Ex: mc = hoomd.hpmc.integrate.Sphere(23456)
+        mc.shape["A"] = {"diameter": 1}
+        check_dict(mc.shape["A"], {"diameter": 1})
+
+    Useful for more complex nested dictionaries (like the shape key in unions)
+    Used to test that the dictionary passed in is what gets passed out
+    """
+    for key, val in args.items():
+        if isinstance(shape_dict[key], list) and len(shape_dict[key]) > 0 \
+           and key != 'shapes':
+            np.testing.assert_allclose(shape_dict[key], val)
+        elif key == 'shapes':
+            for shape_args in shape_dict[key]:
+                for shape_key, shape_val in shape_args.items():
+                    if isinstance(shape_args[shape_key], list) \
+                       and len(shape_args[shape_key]) > 0:
+                        np.testing.assert_allclose(shape_args[shape_key],
+                                                   shape_val)
+                    else:
+                        assert shape_args[shape_key] == shape_val
+        else:
+            np.testing.assert_almost_equal(shape_dict[key], val)
+
+
+def test_dict_conversion(shape_dict_conversion_args):
+    for shape_params, args_list in shape_dict_conversion_args():
+        for args in args_list:
+            test_shape = shape_params(args)
+            test_dict = test_shape.asDict()
+            check_dict(test_dict, args)
+
+
+def test_shape_params(integrator_args):
+    for shape_integrator, valid_args, invalid_args in integrator_args():
+        mc = shape_integrator(23456)
+        for args in valid_args:
+            mc.shape["A"] = args
+            check_dict(mc.shape["A"], args)
+        for args in invalid_args:
+            with pytest.raises(Exception):
```

Comment:
I think this is always the same type of exception isn't it? I can't remember for sure, but I believe it is a `hoomd.typeconverter.TypeConverterError`.

Reply:
Fixed ba04b2da8